### DISCUSSION TITLE
Custom html response page when domain is blocked

### DISCUSSION THREAD

**Discussion by ambipur76** [DNSCrypt/dnscrypt-proxy#1961] [Q&A]
dnscrypt-proxy --check
[2021-12-15 23:58:18] [NOTICE] dnscrypt-proxy 2.0.45
[2021-12-15 23:58:18] [NOTICE] Source [public-resolvers] loaded
[2021-12-15 23:58:18] [NOTICE] Source [relays] loaded
[2021-12-15 23:58:18] [NOTICE] Configuration successfully checked

Operating system: debian buster x86

I am trying a custom html error page when a domain is blocked with the help of the following configuration option:
blocked_query_response      = 'A:192.168.1.1/errors/status-404.html'
Unfortunately the error page does not display when invoking a domain from the blocklist. Is this feasible?

**Comment by jedisct1** [comment]
This is not possible. The DNS protocol cannot serve web pages.

But the fact that this line was accepted by the parser is definitely a bug :)

**Comment by ambipur76** [comment]
I have a working web server on the same machine. Is it not possible to redirect somehow to the web server to serve that custom error page?
As a second option, implementing a minimalistic web server would be a huge effort?

**Comment by ambipur76** [comment]
Indeed, the following configuration option works:
blocked_query_response      = 'A:176.103.130.135'
It opens a friendly blocking page with a picture. However I'd better use my own, it possible.

**Comment by jedisct1** [comment]
Virtually all traffic is HTTPS these days.

**Comment by ambipur76** [comment]
The following configuration options are also accepted by dnscrypt-proxy:
blocked_query_response      = 'A:example.com/dnsblock'
or
blocked_query_response      = 'A:192.168.1.1/dnsblock'
Of course I have used my own domain name instead of example.com. Unfortunately my custom page does not appear in the browser when specifying a subdirectory after the address. Could this be changed?
